src/T2DMap.cpp

```diff
@@ -900,6 +909,15 @@ inline void T2DMap::drawRoom(QPainter& painter, QFont& roomVNumFont, QPen& pen,
         }
 
     }
+    // Do we need to draw the room name:
+    if (mShowRoomName && areRoomIdsLegible) {
+        painter.save();
+        QColor roomIdColor;
+        painter.setPen(QPen(QColor(Qt::white)));
+        painter.setFont(roomVNumFont);
```

Comment:
I'd check whether this is a good font instance to use - IIRC I adjust it's size dynamically so that the largest room number in the currently shown area will just fit within the room square (so will be different if all the rooms in the area are less than, say, 10,000 compared to maps with room numbers in the millions) - so the corresponding room name text will be dependent on that as well.

You'd be better advised to work out the maximum length (remember the room names are not confined to ASCII characters so counting the `QChar`s in the text is NOT sufficient) to determine how much space the text will take up on screen - and adjust the font size appropriately! Also the font used for the room numbers may not be adequate to show all room names...

Reply:
I'l add a separate option for room names and I'll figure out more reasonable values for the font sizes.